### PR TITLE
Document new Agent option (hostname_trust_uts_namespace)

### PR DESCRIPTION
**Pull request by vboulineau**
### What does this PR do?

Document new hostname option introduced in https://github.com/DataDog/datadog-agent/pull/14569

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/agent/troubleshooting/hostname_containers.md [hunk 1]
```diff
@@ -215,21 +215,27 @@ Accessing the cloud provider metadata endpoint allows Datadog to properly match
 Encountering this issue usually means that access to the metadata endpoint has been restricted.
 For example, on AWS, this could be due to the [hop limit setting][5].
 
-## Hostname errors in CI environments and sidecar setups
+## Hostname errors in CI environments, sidecar setups, environments without access to container runtime
```
Comment (rtrieu) [inline content/en/agent/troubleshooting/hostname_containers.md:218]:
```suggestion
## Hostname errors in CI environments, sidecar setups, and environments without access to container runtime
```

File: content/en/agent/troubleshooting/hostname_containers.md [hunk 2]
```diff
@@ -215,21 +215,27 @@ Accessing the cloud provider metadata endpoint allows Datadog to properly match
...
 
 When you run the Agent in a **CI environment** (so Agent is ephemeral) or as a sidecar without access to
-host information, set `DD_HOSTNAME` to a value:
+host information, two options are available:
+
+- Setting `DD_HOSTNAME` (`hostname` in `datadog.yaml`) explicitly to host hostname:
```
Comment (rtrieu) [inline content/en/agent/troubleshooting/hostname_containers.md:223]:
```suggestion
- Setting `DD_HOSTNAME` (`hostname` in `datadog.yaml`) explicitly to the hostname:
```

File: content/en/agent/troubleshooting/hostname_containers.md [hunk 3]
```diff
@@ -215,21 +215,27 @@ Accessing the cloud provider metadata endpoint allows Datadog to properly match
...
+- Setting `DD_HOSTNAME_TRUST_UTS_NAMESPACE` (`hostname_trust_uts_namespace` in `datadog.yaml`):
+
+This option is available starting Datadog Agent **7.42.0**.
 
 ```
--e DD_HOSTNAME=<my_hardcoded_hostname>
+-e DD_HOSTNAME_TRUST_UTS_NAMESPACE=true
 ```
 
+When this is set, the Agent will use in-container hostname (usually, container name or pod name).
```
Comment (rtrieu) [inline content/en/agent/troubleshooting/hostname_containers.md:237]:
```suggestion
When this is set, the Agent will use the in-container hostname (usually the container name or pod name).
```

File: content/en/agent/troubleshooting/hostname_containers.md [hunk 4]
```diff
@@ -215,21 +215,27 @@ Accessing the cloud provider metadata endpoint allows Datadog to properly match
...
+
 **Note:** This does not apply to serverless solutions like Fargate.
```
Comment (rtrieu) [inline content/en/agent/troubleshooting/hostname_containers.md:239]:
```suggestion
**Note**: This does not apply to serverless solutions like Fargate.
```

**Review by rtrieu — CHANGES_REQUESTED** [review]
Left some minor suggestions for your review. Otherwise LGTM!

**Review by rtrieu — APPROVED** [review]
Thanks for making the changes!

**Comment by vboulineau** [comment]
@rtrieu Thanks for the review, I don't have the rights to merge so I'd need for you to do it :)

**Pull request merged**